grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1753 @@
+                    else:
+                        manual_standardized_score = np.round(
+                            (cm_reward - cm_reward_min_ep) / (cm_reward_max_ep - cm_reward_min_ep + 1e-5), 4)
+                        manual_score = DEFAULT_PARAMS_TRUSTSCORE["min_score"] + (
+                                    env._reward_helper.template_reward.max_score - DEFAULT_PARAMS_TRUSTSCORE[
+                                "min_score"]) * manual_standardized_score
+
+                        assert score == manual_score
+                else : 
+                    assert score == 0
+
+    def test_assistant_trust_score_no_blackout_2_attack_same_time_2_alert(self) -> None :
+        """ When we raise 2 alerts for 2 attacks at the same time (step 2)
+            but no blackout occur, we expect a score of 41
+        """
+        kwargs_opponent = dict(lines_attacked=[ATTACKED_LINE]+['48_53_141'], 
+                                   duration=3, 
+                                   steps_attack=[2])
+        with make(self.env_nm,
+                  test=True,
+                  difficulty="1", 
+                  reward_class=_AlertTrustScore(**DEFAULT_PARAMS_TRUSTSCORE),
+                  opponent_attack_cooldown=0, 
+                  opponent_attack_duration=99999, 
+                  opponent_budget_per_ts=1000, 
+                  opponent_init_budget=10000., 
+                  opponent_action_class=PlayableAction, 
+                  opponent_class=TestOpponent, 
+                  kwargs_opponent=kwargs_opponent,
+                  _add_to_name="_tatsnb2ast2a"
+            ) as env : 
+            env.seed(0)
+            env.reset()
+            step = 0
+            for i in range(env.max_episode_duration()):
+                attackable_line_ids = [0, 1]
+                act = env.action_space()
+                if step == 1 :
+                    act = env.action_space({"raise_alert": attackable_line_ids})
+                obs, score, done, info = env.step(act)
+                step += 1
+                if step in _get_steps_attack(kwargs_opponent): 
+                    assert info["opponent_attack_line"] is not None, f"no attack is detected at step {step}"
+                else:
+                    assert info["opponent_attack_line"]  is None, f"an attack is detected at step {step}"
+                    
+                if done:
+
+                    total_nb_attacks = env._reward_helper.template_reward.total_nb_attacks
+                    nb_last_attacks = env._reward_helper.template_reward.nb_last_attacks
+
+                    assert nb_last_attacks == 0
+                    assert total_nb_attacks == 2
+
+                    cm_reward=env._reward_helper.template_reward.cumulated_reward
+                    assert env._reward_helper.template_reward.cumulated_reward==DEFAULT_PARAMS_TRUSTSCORE["reward_end_episode_bonus"] +\
+                            DEFAULT_PARAMS_TRUSTSCORE["reward_min_no_blackout"]*total_nb_attacks
```

Comment:
this assert should be : `env._reward_helper.template_reward.cumulated_reward==DEFAULT_PARAMS_TRUSTSCORE["reward_end_episode_bonus"] + DEFAULT_PARAMS_TRUSTSCORE["reward_min_no_blackout"]`

Reply:
Yes no_blackout_2_attack_same_time tests are updated in the next commit I will do. This relates to how we decide to count the number of attacks which required a change in _alertTrustScore as well